### PR TITLE
closes #13: Implement caching service

### LINKED ISSUE #13: Implement caching service
Implement a caching service that can cache lists, movies, and users. The caching service can have get, set, and delete. We'll have to decide where to place these guards.

### PR DESCRIPTION
**Pull request by manekenpix**
## Issue

closes #13 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This  is a first implementation of caching for lists and movies. In a follow-up PR, it can be abstracted to its own service, and it'll include caching users.

## Testing the PR
- Make sure there's a way to check the content of a redis db.
- Start the redis db using the new changes in `docker-compose.yml`.
- Assert that fetching movies before and after using cache return correct responses.

## Checklist

- [x] **Quality**: This PR builds and passes the respective npm run test and works locally
- [ ] **Tests**: This PR includes thorough tests
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: api/package-lock.json
Comment (chrispinkney) [inline api/package-lock.json:1]: Do we want to commit package lock?

File: docker-compose.yml [hunk 1]
```diff
@@ -45,3 +45,25 @@ services:
     depends_on:
       - api
     restart: "unless-stopped"
+
+  redis:
```
Comment (chrispinkney) [inline docker-compose.yml:49]: Do we also want the app to halt if redis is down/broken for both prod and dev? I think we panic if the db isn't running when bootstrapping the backend, do we want the same behavior for redis? Just wondering.

**Review by chrispinkney — APPROVED** [review]
Really cool stuff here. App seems to be working as intended too, woohoo.

File: docker-compose.yml (hunk 1 above)
Comment (manekenpix) [reply docker-compose.yml:49]: I'll address that in a follow-up PR.

**Pull request merged**